### PR TITLE
Run macos applescript in fork

### PR DESCRIPTION
**Pull request by iloveitaly**
unfortunately, it's well-documented that applescript leaks memory like crazy:

- https://macscripter.net/viewtopic.php?id=41564
- https://github.com/Hammerspoon/hammerspoon/issues/1980

It's not designed to be executed over and over by the same process. In our case, this
resulted in massive memory usage over time. To work around this, this forks a process
to run our applescript and kill it every so often to clean out it's memory usage.

It's not pretty, but neither is 1GB memory usage for the window watcher. @ErikBjare I'm curious your take on this approach and if I should clean it up for merging.

### CODE REVIEW

File: aw_watcher_window/macos_jxa.py [hunk 1]
```diff
@@ -21,17 +23,20 @@ def compileScript():
...
         os.path.dirname(os.path.realpath(__file__)), "printAppStatus.jxa"
     )
+
     with open(scriptPath) as f:
         scriptContents = f.read()
 
-        # remove shebang line
-        if scriptContents.split("\n")[0].startswith("#"):
-            scriptContents = "\n".join(scriptContents.split("\n")[1:])
+        # # remove shebang line
```
Comment (iloveitaly) [inline aw_watcher_window/macos_jxa.py:37]: do you remember why this was needed? I'm not running into any errors with this removed.

File: aw_watcher_window/__main__.py [hunk 2]
```diff
@@ -1,3 +1,6 @@
 from aw_watcher_window import main
 
-main()
+# without this guard, running __main__.py directly will result in a multiprocess
+# related error on macos. This makes it challenging to debug the window watcher on macos
+if __name__ == "__main__":
+  main()
```
Comment (ErikBjare) [inline aw_watcher_window/__main__.py:6]:
```suggestion
    main()
```

File: aw_watcher_window/macos_jxa.py [hunk 3]
```diff
@@ -44,9 +49,109 @@ def compileScript():
 
     return script
 
+def wrapped_function(condition, result_queue: multiprocessing.Queue, function_reference: t.Callable) -> None:
+    # this is run in a forked process, which wipes all logging configuration
```
Comment (ErikBjare) [inline aw_watcher_window/macos_jxa.py:53]: Forking wipes logging config? :O

File: aw_watcher_window/main.py [hunk 4]
```diff
@@ -15,12 +15,6 @@
 
 logger = logging.getLogger(__name__)
 
-# run with LOG_LEVEL=DEBUG
-log_level = os.environ.get("LOG_LEVEL")
-if log_level:
-    logger.setLevel(logging.__getattribute__(log_level.upper()))
-
-
```
Comment (ErikBjare) [inline aw_watcher_window/main.py:23]: Why was this removed?

File: aw_watcher_window/macos_jxa.py (hunk 3 above)
Comment (iloveitaly) [reply aw_watcher_window/macos_jxa.py:53]: Seems to. I think it has something to do with the logging config tied to file handles which aren't valid in the forked process? Unsure, was hoping you/someone else understood this more.

File: aw_watcher_window/main.py (hunk 4 above)
Comment (iloveitaly) [reply aw_watcher_window/main.py:23]: This is now in aw-client https://github.com/ActivityWatch/aw-client/commit/4b00b98bb39114aedb819e113a7e85f05027024b

**Comment by iloveitaly** [comment]
@ErikBjare I just tried from source. I'll build the app and see it things continue to work.

Question for you: what are your thoughts on increasing poll time to 3s? Polling via applescript takes up a non-trivial amount of CPU time and 1s resolution doesn't get me anything vs 3s (or 10s in my case).

File: aw_watcher_window/macos_jxa.py (hunk 3 above)
Comment (ErikBjare) [reply aw_watcher_window/macos_jxa.py:53]: File handle sounds like a likely culprit. But, just to make sure, "wipes *all* logging configuration" isn't true, is it?

**Comment by iloveitaly** [comment]
@ErikBjare with the new swift-based notifications in place, we won't need this. At some point we should rip out the python-based tracker for macOS as well.
